internal/compiler/passes/const_propagation.rs

```diff
@@ -148,6 +161,38 @@ fn simplify_expression(expr: &mut Expression) -> bool {
             }
             can_inline
         }
+        Expression::Condition { condition, true_expr, false_expr } => {
+            let mut can_inline = simplify_expression(condition);
+            can_inline &= match &**condition {
+                Expression::BoolLiteral(true) => {
+                    *expr = *true_expr.clone();
+                    simplify_expression(expr)
+                }
+                Expression::BoolLiteral(false) => {
+                    *expr = *false_expr.clone();
+                    simplify_expression(expr)
+                }
+                _ => simplify_expression(true_expr) && simplify_expression(false_expr),
+            };
+            can_inline
+        }
+        Expression::CodeBlock(stmts) if stmts.len() == 1 => {
+            *expr = stmts[0].clone();
+            simplify_expression(expr)
+        }
+        Expression::FunctionCall { function, arguments, .. } => {
+            let mut args_can_inline = true;
+            for arg in arguments.iter_mut() {
+                args_can_inline &= simplify_expression(arg);
+            }
+            if args_can_inline {
+                if let Some(inlined) = try_inline_function(function, arguments) {
+                    *expr = inlined;
+                    true;
+                }
+            }
+            false
```

Comment:
I think this could be `args_can_inline` 

Even if the function itself cannot be inlined, the expression that contains a function call can still be inlined

... or, maybe on a second thought it depends if this is calling a builtin function or have a reference to function or callback. It is strange that the code was previously returning true and that worked.

Reply:
The previous version was broken in ed179cd5b1ba7a1074d6484da5c228. It returned false before that.

This `false` here is only reached if `try_inline_function` failed, i.e. the function name itself is a non-const binding, or the body contained something non-const (other than the args)

BTW, I forgot to actually return the `true`.

Reply:
Indeed. Thanks for the explanation.